syntax = "proto3";

package speaker_emotion;

// Speaker Emotion service definition.
service SpeakerEmotion {
  rpc Analyze (stream Request) returns (stream Response) {}
}

message Request {
  repeated float signal = 1 [packed=true];
  int32 sample_rate = 2;
}

message Emotion {
  float neutral = 1;
  float calm = 2;
  float happy = 3;
  float sad = 4;
  float angry = 5;
  float fearful = 6;
  float surprise = 7;
  float disgust = 8;
}

message Response {
  Emotion emotion = 1;
  float exec_time = 2;
}
